lib/node_modules/@stdlib/lapack/base/dlaset/README.md

```diff
@@ -0,0 +1,230 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# dlaset
+
+> Initialize an `M` by `N` matrix `A` to `beta` on the diagonal and `alpha` on the off-diagonals.
+
+<section class = "usage">
+
+## Usage
+
+```javascript
+var dlaset = require( '@stdlib/lapack/base/dlaset' );
+```
+
+#### dlaset( order, uplo, M, N, alpha, beta, A, LDA )
+
+Initializes an `M` by `N` matrix `A` to `beta` on the diagonal and `alpha` on the off-diagonals.
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+var A = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
+
+dlaset( 'row-major', 'lower', 3, 3, 3.1, 3.7, A, 3 );
+// A => <Float64Array>[ 3.7, 0.0, 0.0, 3.1, 3.7, 0.0, 3.1, 3.1, 3.7 ]
+```
+
+The function has the following parameters:
+
+-   **order**: storage layout.
+-   **uplo**: specifies whether the upper or lower triangular part of matrix `A` is supplied.
+-   **M**: number of rows in `A`.
+-   **N**: number of columns in `A`.
+-   **alpha**: scalar constant to which off-diagonal elements are set.
+-   **beta**: scalar constant to which diagonal elements are set.
+-   **A**: input [`Float64Array`][mdn-float64array] matrix `A`.
+-   **LDA**: stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`).
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments, max-len -->
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+// Initial arrays...
+var A0 = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
+
+// Create offset views...
+var A1 = new Float64Array( A0.buffer, A0.BYTES_PER_ELEMENT*1 ); // start at 1st element
+
+dlaset( 'row-major', 'lower', 3, 3, 3.1, 3.7, A1, 3 );
+// A0 => <Float64Array>[ 0.0, 3.7, 0.0, 0.0, 3.1, 3.7, 0.0, 3.1, 3.1, 3.7 ]
+```
+
+#### dlaset.ndarray( order, uplo, M, N, alpha, beta, A, LDA, offsetA )
```

Comment:
@Pranavchiku Would you minding updating the implementation to support separate dimension strides, as discussed elsewhere?

Reply:
On it, thus marking it as draft.